lib/engine/game/g_1880/game.rb

```diff
@@ -650,6 +664,29 @@ def check_for_foreign_investor_connection(fi)
           graph = Graph.new(self, no_blocking: true)
           graph.reachable_hexes(corporation).include?(fi_home_token)
         end
+
+        def after_par(corporation)
+          super
+          @par_order[corporation.par_price.price] << corporation
+        end
+
+        def operating_order
+          @minors.select(&:floated?) + @par_order.values.flatten
+        end
+
+        def after_buying_train(train)
+          return if train.name == @depot.upcoming.first.name
+
```

Comment:
this check will return false if cross buying trains, or buying trains that don't trigger train removal (20, 8e,10) this was already fixed in my local dev branch, will be included in a future PR